internal/storage/v2/elasticsearch/tracestore/to_dbmodel_test.go

```diff
@@ -336,6 +339,59 @@ func TestToDBModel(t *testing.T) {
 	}
 }
 
+func TestToDbModel_Fixtures(t *testing.T) {
+	span := getDbSpan(t)
+	td, err := FromDBModel([]dbmodel.Span{span})
+	require.NoError(t, err)
+	assertTraces(t, td)
+	spans := ToDBModel(td)
+	assertSpans(t, spans)
+}
+
+func writeActualData(t *testing.T, name string, data []byte) {
+	var prettyJson bytes.Buffer
+	err := json.Indent(&prettyJson, data, "", "  ")
+	require.NoError(t, err)
+	err = os.WriteFile("actual_"+name+".json", prettyJson.Bytes(), 0o644)
+	require.NoError(t, err)
+}
+
+func assertTraces(t *testing.T, td ptrace.Traces) {
+	if !assert.Equal(t, getPTraces(t), td) {
+		marshaller := ptrace.JSONMarshaler{}
+		data, err := marshaller.MarshalTraces(td)
+		require.NoError(t, err)
+		writeActualData(t, "traces", data)
+	}
+}
+
+func assertSpans(t *testing.T, spans []dbmodel.Span) {
+	assert.Len(t, spans, 1)
+	if !assert.Equal(t, getDbSpan(t), spans[0]) {
+		jsonData, err := json.Marshal(spans[0])
+		require.NoError(t, err)
+		writeActualData(t, "spans", jsonData)
+	}
+}
+
+func getDbSpan(t *testing.T) dbmodel.Span {
+	dbStr, err := os.ReadFile("fixtures/es_01.json")
```

Comment:
You are spreading responsibility all over. The test knows which fixtures it's using, not the helper function.